libs/test/SmartSyncTest/src/com/salesforce/androidsdk/smartsync/manager/SyncManagerTest.java

```diff
@@ -282,6 +293,31 @@ public void testSyncUpWithLocallyDeletedRecords() throws Exception {
         checkServerDeleted(idsLocallyDeleted);
 	}
 
+    /**
+     * Create accounts locally, delete them locally, sync up with merge mode LEAVE_IF_CHANGED, check smartstore
+     *
+     * Ideally an application that deletes locally created records should simply remove them from the smartstore
+     * But if records are kept in the smartstore and are flagged as created and deleted (or just deleted), then
+     * sync up should not throw any error and the records should end up being removed from the smartstore
+     *
+     */
+    public void testSyncUpWithLocallyCreatedAndDeletedRecords() throws Exception {
```

Comment:
If you revert the SyncUpTarget.java change above, this test will fail with the JSONException reported in the bug.